packages/core/src/new-api/internal/reconciliation/execution-state-resolver.ts

```diff
@@ -26,22 +27,25 @@ export class ExecutionStateResolver {
     constructorArgs: ArgumentType[],
     context: ReconciliationContext
   ): ArgumentType[] {
-    return replaceWithinArg<ArgumentType>(constructorArgs, {
-      bigint: (bi) => bi.toString(),
-      future: (f) => {
-        if (!isContractFuture(f)) {
-          throw new IgnitionError(
-            `Only deployable contract and library futures can be used in args, ${f.id} is not a contract or library future`
-          );
-        }
-
-        return ExecutionStateResolver.resolveContractToAddress(f, context);
-      },
-      accountRuntimeValue: (arv) => context.accounts[arv.accountIndex],
-      moduleParameterRuntimeValue: (mprv) => {
-        return resolveModuleParameter(mprv, context);
-      },
-    }) as ArgumentType[];
+    const replace = (arg: ArgumentType) =>
+      replaceWithinArg<ArgumentType>(arg, {
+        bigint: (bi) => bi.toString(),
+        future: (f) => {
+          if (!isContractFuture(f)) {
+            throw new IgnitionError(
+              `Only deployable contract and library futures can be used in args, ${f.id} is not a contract or library future`
+            );
+          }
+
+          return ExecutionStateResolver.resolveContractToAddress(f, context);
+        },
+        accountRuntimeValue: (arv) => context.accounts[arv.accountIndex],
+        moduleParameterRuntimeValue: (mprv) => {
+          return resolveModuleParameter(mprv, context);
+        },
+      });
+
+    return constructorArgs.map(replace);
```

Comment:
I left a comment about the `as ArgumentType` in a previous commit. It's cool that TS can infer it like this.